docs/docker/README.md

```diff
@@ -4,100 +4,134 @@
+Enter your public hostname
 
-   ```
-   HostIdentity[keymaster.DOMAIN]:keymaster.mydomain.com
-   ```
+```
+HostIdentity[keymaster.DOMAIN]:keymaster.mydomain.com
+```
 
-   Hit enter for the ports and accept the default.
+Hit enter for the ports and accept the default.
 
-   ```
-   HttpAddress[:443]:
-   AdminAddress[:6920]:
-   ```
+```
+HttpAddress[:443]:
+AdminAddress[:6920]:
+```
 
 Fix the config issues mentioned in the main README.md
 
-   ```
-   $ sudo sed -i 's% data_directory:.*% data_directory: "/var/lib/keymaster"%g' \
-    ${KEYMASTER_DATA}/conf/config.yml
-   $ sudo sed -i 's% shared_data_directory:.*% shared_data_directory: "/usr/share/keymasterd/"%g' \
-    ${KEYMASTER_DATA}/conf/config.yml
-   ```
+```
+$ sudo sed -i 's% data_directory:.*% data_directory: "/var/lib/keymaster"%g' \
+ ${KEYMASTER_DATA}/conf/config.yml
+$ sudo sed -i 's% shared_data_directory:.*% shared_data_directory: "/usr/share/keymasterd/"%g' \
+ ${KEYMASTER_DATA}/conf/config.yml
+```
 
 ## Start
 
 After bootstrapping configs and keys you just start the container. This will
 start it sealed.
 
-   ```
-   $ docker-compose up -d
-   ```
+```
+$ docker-compose up -d
+```
 
 ## Unseal
 
 By default the CA will start in a sealed state. To unseal it you will need to
 enter your passphrase.
 
-   ```
-   $ docker exec -e SSL_CERT_FILE=/etc/keymaster/server.pem -it keymaster \
-    /app/keymaster-unlocker -cert /etc/keymaster/adminClient.pem \
-     -key /etc/keymaster/adminClient.key -keymasterHostname localhost
-   Password for unlocking localhost: 
-   OK
-   ```
+```
+$ docker exec -e SSL_CERT_FILE=/etc/keymaster/server.pem -it keymaster \
+  /app/keymaster-unlocker -cert /etc/keymaster/adminClient.pem \
+  -key /etc/keymaster/adminClient.key -keymasterHostname localhost
+Password for unlocking localhost: 
+OK
+```
 
 ## Add users
 
 By default a user is created. Let's start by deleting this and creating our own
 user.
 
-   ```
-   $ rm -f ${KEYMASTER_DATA}/conf/passfile.htpass
-   $ docker exec -it keymaster /usr/bin/htpasswd -B -c \
-    /etc/keymaster/passfile.htpass $USERNAME
-   ```
+```
+$ rm -f ${KEYMASTER_DATA}/conf/passfile.htpass
+$ docker exec -it keymaster /usr/bin/htpasswd -B -c \
+ /etc/keymaster/passfile.htpass $USERNAME
+```
+
+## Add users
+
+By default a user is created. Let's start by deleting this and creating our own
+user.
+
+```
+$ rm -f ${KEYMASTER_DATA}/conf/passfile.htpass
+$ docker exec -it keymaster /usr/bin/htpasswd -B -c \
+ /etc/keymaster/passfile.htpass $USERNAME
+```
+
+## SSH
+
+Distribute the SSH CA to hosts
+
+```
+${KEYMASTER_DATA}/conf/masterKey.asc.pub -> destination:/etc/ssh/masterKey.asc.pub
+```
+Configure sshd to trust the CA by adding this line to ```/etc/ssh/sshd_config```
+
+```
+TrustedUserCAKeys /etc/ssh/masterKey.asc.pub
+```
+
+SSH will allow clients who have run the keymaster client to login.
+
+**NOTE** Username must match principal on all hosts or you will need to
+configure AuthorizedPrincipalsFile.
```

Comment:
side comment: I have for work reasons an allow-principals-command that will work both against a git repository.. that way you dont need to redeploy to change memberships. Are you interested on this (parts are closed source for work, but I can probably split it, if there is interest).

Reply:
Sounds like an interesting use case. Would love to know more, it might help me.